CMakeLists.txt

```diff
@@ -164,47 +164,47 @@ if (XCODE)
+  add_compile_options(
+    # Disable unknown pragma warnings (e.g. for #pragma mark).
+    "-wd4068"
+
+    # TODO: these warnings come from llvmSupport. Since we don't want to diverge from llvmSupport by
+    # addressing these warnings , disable these for now to clean the build log.
+    # If/when we move to use LLVM's own llvmSupport, we should reenable these warnings.
+    "-wd4996" # POSIX name for this item is deprecated.
+    "-wd4244" # Signed conversion.
+    "-wd4267" # Possible loss of data conversions.
+    "-wd4291" # Operator new with no matching delete found.
+    "-wd4800" # Forcing value to bool 'true' or 'false'.
+    "-wd4146" # Unary minus applied to unsigned type.
+    "-wd4141" # 'inline' used more than once.
+    "/EHsc")
 else ()
   # Compile with C++14, without RTTI or exceptions.
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -fno-rtti -fno-exceptions")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS}")
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS}")
-  # Enable additional Clang warnings.
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wmost")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wdocumentation  -Woverloaded-virtual")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wparentheses -Wswitch")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wunused-function -Wunused-variable")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wunused-value -Wempty-body")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wuninitialized -Wconstant-conversion")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wint-conversion -Wbool-conversion")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wenum-conversion -Wsign-compare")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnewline-eof -Wdeprecated-declarations")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Winvalid-offsetof")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wimplicit-fallthrough")
+  add_compile_options(
+   "-std=c++14" "-fno-rtti" "-fno-exceptions"
+    # Enable additional Clang warnings.
+   "-Wmost"
+   "-Wdocumentation" "-Woverloaded-virtual"
+   "-Wparentheses" "-Wswitch"
+   "-Wunused-function" "-Wunused-variable"
+   "-Wunused-value" "-Wempty-body"
+   "-Wuninitialized" "-Wconstant-conversion"
+   "-Wint-conversion" "-Wbool-conversion"
+   "-Wenum-conversion" "-Wsign-compare"
+   "-Wnewline-eof" "-Wdeprecated-declarations"
+   "-Winvalid-offsetof"
+   "-Wnon-virtual-dtor"
+   "-Wimplicit-fallthrough")
 endif ()
 
 if (${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DLLVM_ON_WIN32")
+  add_compile_options("-DLLVM_ON_WIN32")
 endif()
 
 # On BSD and Linux, always build with PIC.
 if(${CMAKE_SYSTEM_NAME} MATCHES ".*BSD" OR ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+  add_compile_options("-fPIC")
 endif ()
```

Comment:
Please replace the 3 lines above with `set(CMAKE_POSITION_INDEPENDENT_CODE YES)`